src/sentry/api/endpoints/organization_discover_query.py

```diff
@@ -88,27 +87,32 @@ def __init__(self, *args, **kwargs):
     def validate(self, data):
         data['arrayjoin'] = self.arrayjoin
 
-        return data
-
-    def validate_range(self, attrs, source):
-        has_start = bool(attrs.get('start'))
-        has_end = bool(attrs.get('end'))
-        has_range = bool(attrs.get('range'))
-
-        if has_start != has_end or has_range == has_start:
+        # prevent conflicting date ranges from being supplied
+        date_fields = ['start', 'statsPeriod', 'range', 'statsPeriodStart']
+        date_fields_provided = len([data.get(f) for f in date_fields if data.get(f) is not None])
```

Comment:
Since start and end are now being handled in `get_date_range_from_params` should this validation (apart from maybe the range part) move into that function as well?

Reply:
i generally would prefer to document the order in which we try to use these date filters than return an error for someone sending two along. since the behavior of discover was already more restrictive, i think it's fine to error here, but i'd rather not do that for the other endpoints that are using this